### ISSUE TITLE
remove() is a deprecated function

### ISSUE THREAD

**Issue by oneil213** [bradtraversy/mern_shopping_list#41]
After taking your course on youtube, I used the remove() function and I noticed all the items on the list get deleted. Then I did a little read up and saw that remove() is a deprecated function and has been replaced by deleteOne() (to delete a single document). I updated my code and it was fine. Thought you should know. Thanks!

**Comment by bradtraversy** [comment]
Good catch @oneil213 — `.remove()` was deprecated in Mongoose 7 and fully removed in Mongoose 8. With the recent dependency updates in PR #100, this is likely to become a breaking issue if Mongoose is bumped. The fix is to replace `.remove()` with `.deleteOne()` on the document instance (or use `Item.deleteOne({ _id: id })` at the model level). Worth addressing as part of the dep update work.
